### PR TITLE
Add support for Databricks runtime 14.3 & 15.4 LTS

### PR DESCRIPTION
**Pull request by ebyhr**
## Release notes

(x) This is not user-visible or is docs only, and no release notes are required.

### CODE REVIEW

**Review by findepi — APPROVED** [review]
thanks!

**Comment by ebyhr** [comment]
It seems 14.3 LTS enabled deletion vectors by default: https://github.com/trinodb/trino/actions/runs/8840347798/job/24275694250?pr=21700

**Comment by ebyhr** [comment]
@benbauer-draeger  This PR just adds a new CI environment of Databricks 14.3 LTS. https://github.com/trinodb/trino/pull/22102 to support writing deletion vectors was already merged in version 454.

File: testing/trino-product-tests/src/main/java/io/trino/tests/product/deltalake/TestDeltaLakeWriteDatabricksCompatibility.java [hunk 1]
```diff
@@ -330,7 +331,7 @@ public void testInsertingIntoDatabricksTableWithAddedNotNullConstraint()
         }
     }
 
-    @Test(groups = {DELTA_LAKE_DATABRICKS, PROFILE_SPECIFIC_TESTS})
+    @Test(groups = {DELTA_LAKE_DATABRICKS_104, PROFILE_SPECIFIC_TESTS})
```
Comment (findinpath) [inline testing/trino-product-tests/src/main/java/io/trino/tests/product/deltalake/TestDeltaLakeWriteDatabricksCompatibility.java:334]: Why is this change made?

**Review by findinpath — APPROVED** [review]

File: testing/trino-product-tests/src/main/java/io/trino/tests/product/deltalake/TestDeltaLakeWriteDatabricksCompatibility.java (hunk 1 above)
Comment (ebyhr) [reply testing/trino-product-tests/src/main/java/io/trino/tests/product/deltalake/TestDeltaLakeWriteDatabricksCompatibility.java:334]: The connector doesn't support VACUUM with deletion vector, so I specified the old version here. Added a comment to revert this change in #22809.

**Comment by mosabua** [comment]
Are you sure we dont want a release notes entry for this @ebyhr and @findinpath ?

**Comment by ebyhr** [comment]
@mosabua We don't need a release note entry because this PR changed only tests.

**Comment by mosabua** [comment]
ok .. thanks @ebyhr

**Pull request merged**